api/proto/config/storage.proto

```diff
@@ -193,3 +193,29 @@ message Volume {
   string displayName = 7;  // Optional friendly name echo'ed in info message
   bool clear_text = 8;  // Flag to indicate the volume encryption needed or not
 }
+
+// ZfsDiskState is an operational state of disks as part of ZFS
+enum ZfsDiskState {
+  ZFS_DISK_STATE_UNSPECIFIED = 0; // no configured, do nothing
+  ZFS_DISK_STATE_ONLINE = 1; // included in zfs and online
+  ZFS_DISK_STATE_OFFLINE = 2; // included in zfs and offline
+  ZFS_DISK_STATE_REMOVED = 3; // removed from zfs
```

Comment:
While most of these states are about ZFS there the unused is not specific to ZFS and we will (in the near future) need some AppDedicated config for direct assignment to applications.
Thus I think "ZFS" should be removed from the prefix and instead have
DISK_CONFIG_UNSPECIFIED
DISK_CONFIG_EVEOS (for completeness; the disk EVE is installed on)
DISK_CONFIG_EXT4 (I think we need this to match a system where persist disks is a single ext4)
DISK_CONFIG_ZFS_ONLINE (can potentially drop "ONLINE" from that name)
DISK_CONFIG_ZFS_OFFLINE
DISK_CONFIG_APPDIRECT (for future direct assignment)
DISK_CONFIG_UNUSED (and can later we assigned to the ZFS pool or become APPDIRECT)